### PR TITLE
Merge csharp feature branch into main ai inference branch

### PR DESCRIPTION
**Pull request by trangevi**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/ai/ModelClient/client.tsp [hunk 1]
```diff
@@ -118,3 +118,7 @@ interface Client3 {
   embed is AI.Model.getImageEmbeddings;
   getModelInfo is AI.Model.getModelInfo;
 }
+
+// The following is to update names as necessary for specific programming languages.
+// Done in this file to provide a single point of reference for any renames.
+@@clientName(ModelClient.StreamingChatChoiceUpdate.delta, "InternalStreamingDeltaMessage", "csharp");
```
Comment (dargilco) [inline specification/ai/ModelClient/client.tsp:124]: Change ModelClient to AI.Model

File: specification/ai/ModelClient/models/chat_completions.tsp [hunk 2]
```diff
@@ -103,9 +111,10 @@ alias ChatCompletionsOptions = {
   @doc("""
     ID of the specific AI model to use, if more than one model is available on the endpoint.
     """)
-  @clientName("DeploymentName", "csharp")
   `model`?: string;
-};
+
+  ...Record<unknown>;
```
Comment (dargilco) [inline specification/ai/ModelClient/models/chat_completions.tsp:116]: Shouldn't this be limited to C# only?

File: specification/ai/ModelClient/tspconfig.yaml [hunk 3]
```diff
@@ -23,3 +23,10 @@ options:
     package-dir: "azure-ai-inference"
     package-name: "{package-dir}"
     flavor: azure
+  "@azure-tools/typespec-csharp":
+    package-dir: "Azure.AI.Inference"
+    namespace: "Azure.AI.Inference"
+    clear-output-folder: true
+    model-namespace: false
+    generate-protocol-methods: false
```
Comment (dargilco) [inline specification/ai/ModelClient/tspconfig.yaml:31]: I see "Krzysztof Cwalina" <kcwalina@microsoft.com> is OOF until 7/22... he was asking why we are not generating the protocol methods. I asked if they are needed by all SDKs, and why. I don't think he answered that question (perhaps he left for vacation before he could answer). But anyway, this is where we would enable them. Okay to leave it "false" for now.

File: specification/ai/ModelClient/client.tsp (hunk 1 above)
Comment (dargilco) [reply specification/ai/ModelClient/client.tsp:124]: I think this line can be deleted.

Comment (trangevi) [reply specification/ai/ModelClient/client.tsp]: Done

**Pull request closed without merging**